### ISSUE TITLE
Update tutorial about shuffling before sharding

### ISSUE THREAD

**Issue by ejguan** [pytorch/data#709]
### 📚 The doc issue

The [tutorial](https://pytorch.org/data/beta/tutorial.html#working-with-dataloader) needs to update the actual reason about shuffling before sharding. It's not accurate.
Shuffling before sharding is required to achieve global shuffling rather than only shuffling inside each shard.

**Comment by NivekT** [comment]
As in the following sentence is inaccurate?

> In order for DataPipe sharding to work with DataLoader, we need to add the following. It is crucial to add ShardingFilter after Shuffler to ensure that all worker processes have the same order of data for sharding.

**Comment by ejguan** [comment]
Without shuffler, each worker process has the same order of data as well. So, it's not the essential reason.
